### PR TITLE
Fixed PyTorch MNIST dataset

### PR DESCRIPTION
**Pull request by tgaddair**
See https://github.com/pytorch/vision/issues/3500

### CODE REVIEW

File: .buildkite/gen-pipeline.sh [hunk 1]
```diff
@@ -178,10 +178,12 @@ run_mpi_integration() {
     run_test "${test}" "${queue}" \
       ":tensorflow: MPI Keras MNIST (${test})" \
       "bash -c \"${oneccl_env} \\\$(cat /mpirun_command) python /horovod/examples/keras/keras_mnist_advanced.py\""
+  fi
 
+  if [[ ${test} != *"torch1_2"* ]]; then
```
Comment (EnricoMi) [inline .buildkite/gen-pipeline.sh:183]: We are adding a few more test runs here as we did not run it for tf2 before, now we do. This is expected?

File: test/single/data/expected_buildkite_pipeline.yaml [hunk 2]
```diff
@@ -1000,6 +1000,22 @@ steps:
     automatic: true
   agents:
     queue: cpu
+- label: ':fire: MPI PyTorch MNIST (test-cpu-mpich-py3_8-tf2_4_1-keras2_4_3-torch1_7_1-mxnet1_7_0_p2-pyspark_3_0_1)'
```
Comment (EnricoMi) [inline test/single/data/expected_buildkite_pipeline.yaml:1003]: here are the new tf2 runs of this example

**Review by EnricoMi — APPROVED** [review]
LGTM! Are you expecting the example script to run for tf2 now?

File: test/single/data/expected_buildkite_pipeline.yaml (hunk 2 above)
Comment (tgaddair) [reply test/single/data/expected_buildkite_pipeline.yaml:1003]: I think this is okay, we should make the decision based on the version of PyTorch, not TF.

File: .buildkite/gen-pipeline.sh (hunk 1 above)
Comment (tgaddair) [reply .buildkite/gen-pipeline.sh:183]: Yeah, in my mind if the Docker has TF 2 and PyTorch 1.3, we should still run it, based off the version of PyTorch.

**Pull request merged**